tests/fail/data_race/atomic_write_na_read_race2.stderr

```diff
@@ -4,6 +4,16 @@ error: Undefined Behavior: Data race detected between Atomic Store on thread `<u
 LL |             (&*c.0).store(32, Ordering::SeqCst);
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Data race detected between Atomic Store on thread `<unnamed>` and Read on thread `<unnamed>` at ALLOC
    |
+help: The Atomic Store on thread `<unnamed>` is here
+  --> $DIR/atomic_write_na_read_race2.rs:LL:CC
+   |
+LL | ...   (&*c.0).store(32, Ordering::SeqCst);
+   |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+help: The Read on thread `<unnamed>` is here
+  --> RUSTLIB/core/src/ptr/mod.rs:LL:CC
+   |
+LL | }
+   | ^
```

Comment:
not sure how useful these are without an extra annotation. I'm surprised the spans don't contain expansion information actually. That could be something we need to fix on the rustc side. I'm guessing it's destructor related?

Reply:
This may be because of the lack of overwriting when there are `DUMMY_SP`. We end up picking the span from the last successful use of that race site

Reply:
Pretty sure it was an old bug that I swear I fixed before... we need to update the span for a clock, even if the operation is a race. Failing to do so results in these wacky spans.